packages/repository/test/unit/repository-mixin/repository-mixin.test.ts

```diff
@@ -0,0 +1,114 @@
+// Copyright IBM Corp. 2013,2017. All Rights Reserved.
+// Node module: @loopback/repository
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {expect} from '@loopback/testlab';
+import {
+  RepositoryMixin,
+  juggler,
+  DataSourceConstructor,
+  Class,
+} from '../../../';
+import {Application, Component} from '@loopback/core';
+
+// tslint:disable:no-any
+
+describe('RepositoryMixin', () => {
+  it('mixed class has .repository()', () => {
+    const myApp = new appWithRepoMixin();
+    expect(typeof myApp.repository).to.be.eql('function');
+  });
+
+  it('binds repository from constructor', () => {
+    const myApp = new appWithRepoMixin({
+      repositories: [NoteRepo],
+    });
+
+    expectNoteRepoToBeBound(myApp);
+  });
+
+  it('binds repository from app.repository()', () => {
+    const myApp = new appWithRepoMixin();
+
+    expectNoteRepoToNotBeBound(myApp);
+    myApp.repository(NoteRepo);
+    expectNoteRepoToBeBound(myApp);
+  });
+
+  it('binds user defined component without repository', () => {
+    class EmptyTestComponent {}
+
+    const myApp = new appWithRepoMixin({
+      components: [EmptyTestComponent],
+    });
+
+    expectComponentToBeBound(myApp, EmptyTestComponent);
+  });
+
+  it('binds user defined component with repository in constructor', () => {
+    const myApp = new appWithRepoMixin({
+      components: [TestComponent],
+    });
+
+    expectNoteRepoToBeBound(myApp);
+  });
+
+  it('binds user defined component with repository from .component()', () => {
+    const myApp = new appWithRepoMixin();
+
+    const boundComponentsBefore = myApp.find('components.*').map(b => b.key);
+    expect(boundComponentsBefore).to.be.empty();
+    expectNoteRepoToNotBeBound(myApp);
+
+    myApp.component(TestComponent);
+
+    expectComponentToBeBound(myApp, TestComponent);
+    expectNoteRepoToBeBound(myApp);
+  });
+
+  const appWithRepoMixin = RepositoryMixin(Application);
```

Comment:
I find it weird to have a class starting with lower-case letter. I am proposing the following:

```ts
class AppWithRepoMixin extend RepositoryMixin(Application) {
}
```

I think that should be the recommended usage for our users too.